numpy/lib/function_base.py

```diff
@@ -76,150 +76,191 @@ def iterable(y):
-        weights are not supported, so this field must be empty or None.
+    x : array_like
+        Input data that is to be histogrammed.
+
+    Returns
+    -------
+    n : An estimate of the optimal bin count for the given data.
     """
-    if a.size == 0:
-        return 1
+    return int(np.ceil(np.sqrt(x.size)))
 
-    if data_weights is not None:
-        raise TypeError("Automated estimation of the number of "
-                        "bins is not supported for weighted data")
 
-    if data_range is not None:
-        mn, mx = data_range
-        keep = (a >= mn)
-        keep &= (a <= mx)
-        if not np.logical_and.reduce(keep):
-            a = a[keep]
+def _hist_bin_sturges(x):
+    """
+    Sturges histogram bin estimator.
 
-    def sqrt(x):
-        """
-        Square Root Estimator
+    A very simplistic estimator based on the assumption of normality of
+    the data. This estimator has poor performance for non-normal data,
+    which becomes especially obvious for large data sets. The estimate
+    depends only on size of the data.
 
-        Used by many programs for its simplicity.
-        """
-        return np.ceil(np.sqrt(x.size))
+    Parameters
+    ----------
+    x : array_like
+        Input data that is to be histogrammed.
 
-    def sturges(x):
-        """
-        Sturges Estimator
+    Returns
+    -------
+    n : An estimate of the optimal bin count for the given data.
+    """
+    return int(np.ceil(np.log2(x.size))) + 1
 
-        A very simplistic estimator based on the assumption of normality
-        of the data. Poor performance for non-normal data, especially
-        obvious for large ``x``. Depends only on size of the data.
-        """
-        return np.ceil(np.log2(x.size)) + 1
 
-    def rice(x):
-        """
-        Rice Estimator
+def _hist_bin_rice(x):
+    """
+    Rice histogram bin estimator.
 
-        Another simple estimator, with no normality assumption. It has
-        better performance for large data, but tends to overestimate
-        number of bins. The number of bins is proportional to the cube
-        root of data size (asymptotically optimal). Depends only on size
-        of the data.
-        """
-        return np.ceil(2 * x.size ** (1.0 / 3))
+    Another simple estimator with no normality assumption. It has better
+    performance for large data than Sturges, but tends to overestimate
+    the number of bins. The number of bins is proportional to the cube
+    root of data size (asymptotically optimal). The estimate depends
+    only on size of the data.
 
-    def scott(x):
-        """
-        Scott Estimator
+    Parameters
+    ----------
+    x : array_like
+        Input data that is to be histogrammed.
 
-        The binwidth is proportional to the standard deviation of the
-        data and inversely proportional to the cube root of data size
-        (asymptotically optimal).
-        """
-        h = (24 * np.pi**0.5 / x.size)**(1.0 / 3) * np.std(x)
```

Comment:
Thanks for pointing that out. I changed it to this and now apparently clobbered it back. And reverted it again now. I traced the origins of the "3.5" from the [Wikipedia article](https://en.wikipedia.org/wiki/Histogram#Number_of_bins_and_width) in [this paper](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.30.9725) to the formula [`cbrt(24 * sqrt(pi))`](http://www.wolframalpha.com/input/?i=cbrt%2824*sqrt%28pi%29%29) (formula for h-hat-nought on page 5).